autoarray/dataset/interferometer/interferometer.py

```diff
@@ -0,0 +1,218 @@
+        WTildeInterferometer
+            Precomputed values used for the w tilde formalism of linear algebra calculations.
+        """
+
+        from autoarray.inversion.inversion import inversion_util_secret
+
+        logger.info("INTERFEROMETER - Computing W-Tilde... May take a moment.")
+
+        curvature_preload = inversion_util_secret.w_tilde_curvature_preload_interferometer_from(
+            noise_map_real=self.noise_map.real,
+            uv_wavelengths=self.uv_wavelengths,
+            shape_masked_pixels_2d=self.transformer.grid.mask.shape_native_masked_pixels,
+            grid_radians_2d=self.transformer.grid.mask.unmasked_grid_sub_1.in_radians.native,
+        )
+
+        w_matrix = inversion_util_secret.w_tilde_via_preload_from(
+            w_tilde_preload=curvature_preload,
+            native_index_for_slim_index=self.real_space_mask.native_index_for_slim_index,
+        )
+
+        dirty_image = self.transformer.image_from(
+            visibilities=self.visibilities.real * self.noise_map.real**-2.0
+            + 1j * self.visibilities.imag * self.noise_map.imag**-2.0,
+            use_adjoint_scaling=True,
+        )
+
+        return WTildeInterferometer(
+            w_matrix=w_matrix,
+            curvature_preload=curvature_preload,
+            dirty_image=dirty_image,
+            real_space_mask=self.real_space_mask,
+            noise_map_value=self.noise_map[0],
+        )
+
+    @property
+    def mask(self):
+        return self.real_space_mask
+
+    @property
+    def visibilities(self):
+        return self.data
+
+    @property
+    def amplitudes(self):
+        return self.visibilities.amplitudes
+
+    @property
+    def phases(self):
+        return self.visibilities.phases
+
+    @property
+    def uv_distances(self):
+        return np.sqrt(
+            np.square(self.uv_wavelengths[:, 0]) + np.square(self.uv_wavelengths[:, 1])
+        )
+
+    @property
+    def dirty_image(self):
+        return self.transformer.image_from(visibilities=self.visibilities)
+
+    @property
+    def dirty_noise_map(self):
+        return self.transformer.image_from(visibilities=self.noise_map)
+
+    @property
+    def dirty_signal_to_noise_map(self):
+        return self.transformer.image_from(visibilities=self.signal_to_noise_map)
+
+    @property
+    def dirty_inverse_noise_map(self):
+        return self.transformer.image_from(visibilities=self.inverse_noise_map)
+
+    @property
+    def signal_to_noise_map(self):
+
+        signal_to_noise_map_real = np.divide(
+            np.real(self.data), np.real(self.noise_map)
+        )
+        signal_to_noise_map_real[signal_to_noise_map_real < 0] = 0.0
+        signal_to_noise_map_imag = np.divide(
+            np.imag(self.data), np.imag(self.noise_map)
+        )
+        signal_to_noise_map_imag[signal_to_noise_map_imag < 0] = 0.0
+
+        return signal_to_noise_map_real + 1j * signal_to_noise_map_imag
+
+    @property
+    def blurring_grid(self):

```

Comment:
Would an empty array be an elegant choice? Something which if operated on does nothing?